debugger/src/main/java/com/sun/tools/jdi/EnhancedStackFrameImpl.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+// NOSONAR - squid:S1191 - Using the Java debugger API
+package com.sun.tools.jdi; // NOSONAR
+
+import com.sun.jdi.EnhancedStackFrame; // NOSONAR
+import com.sun.jdi.InternalException;
+import com.sun.jdi.InvalidSlotException;
+import com.sun.jdi.InvalidStackFrameException; // NOSONAR
+import com.sun.jdi.Location; // NOSONAR
+import com.sun.jdi.Value; // NOSONAR
+import com.sun.jdi.VirtualMachine; // NOSONAR
+import java.lang.reflect.Field;
+
+/** Implementation of the {@Link EnhancedStackFrame} interface. */
+@SuppressWarnings("squid:S2160" /* only designed to add methods; super.equals() is still valid */)
+public class EnhancedStackFrameImpl extends StackFrameImpl implements EnhancedStackFrame {
```

Comment:
ℹ️ This class is a more generic version of the `ReadDebugIndex` class. By putting the class in the same package as the jdi implementation, we can take advantage of the `StackFrameImpl` base class and avoid doing reflection most of the time.